src/dmd/typesem.d

```diff
@@ -1650,13 +1650,42 @@ private extern (C++) final class TypeSemanticVisitor : Visitor
         import dmd.traits : semanticTraits;
 
         result = null;
+        if (mtype.ty == Terror)
+        {
+            result = mtype;
+            return;
+        }
+        if (mtype.exp.ident != Id.getMember
```

Comment:
@thewilsonator I think this is the special casing @Geod24 was taking about

Reply:
This doesn't support _just_ `getMember`, look below.

Reply:
Yes, there's no problem here. Since Geod24 review i've added support for all the traits that return single symbol or tuple of symbols.

Reply:
you've replaced one special case with multiple special cases.

Reply:
No this covers all the `__traits` that return `alias`able symbols.

Reply:
@UplinkCoder: there's still no problem, the grammar allows all the traits, the semantic restrict them. It's perfectly valid, i even remember [having be told this  here in the past after asking](https://github.com/dlang/dmd/pull/6570#issuecomment-282591380). It's just like a free function that's `const`. It's allowed by the grammar but it trigger an error during the semantic pass. Exact same principle.

Also don't forget that when @Geod24 made his review, the restriction was in the parser, which was indeed an error.

Reply:
Yes of course, this is non-the-less a grammar change.

Reply:
Yes and the dlang.org pull request [has been merged](https://github.com/dlang/dlang.org/pull/2509/files#diff-6e631246d309aefb33fb0aa648c909d8R98): it's a very small change you see.

Finally, I'd like to say that i know that this feature will be on master probably because of a misunderstanding but it's been a week now. Also it's not on yet picked to stable... in case someone would really be late at reading its notifications (sorry guys but if you come at 2:00 AM for a new year party don't be surprised if you have the feeling of having missed the thing), there's still this last protection.